### PR TITLE
Big speedup on reverse dependency calculation

### PR DESCRIPTION
**Pull request by AltGr**
We need to revert a huge simply-linked structure; the big cost was allocating 
the intermediate data, so instead of creating the graph or going through sets, 
we use the integer identifiers already defined by Cudf and store that in a 
simple table.

### CODE REVIEW

**Review by kit-ty-kate — APPROVED** [review]
Indeed! I can confirm an almost 2x speedup on `--depends-on` + `--recursive`.
e.g. `opam list --depends-on dune --recursive -s --all-versions` take
* 9.4s before this PR
* 5.1s with this PR

And no difference in the result :tada:

No noticable difference could be obversed with non-recursive queries

File: src/solver/opamSolver.ml [hunk 1]
```diff
@@ -90,7 +90,14 @@ let cudf_versions_map universe packages =
     pmap OpamPackage.Map.empty
 
 let name_to_cudf name =
-  Dose_common.CudfAdd.encode (OpamPackage.Name.to_string name)
+  let name_s = OpamPackage.Name.to_string name in
+  if String.for_all (function
```
Comment (kit-ty-kate) [inline src/solver/opamSolver.ml:94]: String.for_all is only available since OCaml 4.13. We probably need to introduce `OpamStd.String.for_all`

**Comment by rjbou** [comment]
All these `oppam list` PRs highlighted that there is no opam list test... Feel free to add one, or i can add it later

**Comment by kit-ty-kate** [comment]
Done in https://github.com/ocaml/opam/pull/5006

**Comment by AltGr** [comment]
Test failure doesn't seem related to the PR :confused:

**Pull request merged**